Classes/Utility/GeneralUtility.php

```diff
@@ -413,19 +413,21 @@ public static function fullUrl($url, $domain = null)
             if ($url === '/') {
                 $url = '';
             }
-
-            // remove first /
-            if (strpos($url, '/') === 0) {
-                $url = substr($url, 1);
-            }
-
+            
             if ($domain !== null) {
-                // specified domain
-                $url = 'http://' . $domain . '/' . $url;
+                if(!preg_match('/^https?:\/\//i', $domain)) {
+                    $domain = 'http://' . $domain;
```

Comment:
Well, we have to make some assumptions. If the passed domain hast a scheme, everything is ok, so no scheme is prepended. If there is no scheme - so the regex doesn't match - I think we can't reliably guess the correct scheme at this code line (because in fact, the domain could be any domain), so the best is to stick with http ;-)

Instead of the regex a parse_url call could be used to check, if there is a scheme. Or we could make the regex more general (`'/[a-z]+:\/\//i'`).